internal/util/search/search.go

```diff
@@ -182,3 +209,34 @@ func (sr *SearchReplace) putLiteral(object *yaml.RNode) error {
 func (sr *SearchReplace) shouldPutLiteralByPath() bool {
 	return isAbsPath(sr.ByPath) && sr.ByValue == "" && sr.ByValueRegex == "" && sr.PutLiteral != ""
 }
+
+// settersValues returns the values for the setters present in PutPattern,
+// returns error if PutPattern is invalid or setter value is not a valid literal
+func (sr *SearchReplace) settersValues() (map[string]string, error) {
+	// extract setter name tokens from pattern enclosed in ${}
+	re := regexp.MustCompile(`\$\{([^}]*)\}`)
+	markers := re.FindAllString(sr.PutPattern, -1)
+	if len(markers) == 0 {
+		return nil, errors.Errorf("unable to find setter names in pattern, " +
+			"setter names must be enclosed in ${}")
+	}
+	res := make(map[string]string)
+	for _, marker := range markers {
+		name := strings.TrimSuffix(strings.TrimPrefix(marker, "${"), "}")
+		if setters.DefExists(sr.PackagePath, name) {
+			val, err := setters.SetterValue(name, filepath.Join(sr.PackagePath, kptfile.KptFileName))
```

Comment:
We do we need to look up the setter value from file here? I would expect the information in the Kptfile to be available from the openapi.